File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.js
function solution(word) {
    let set = new Set();
    let vowel= ['A', 'E', 'I', 'O', 'U'];
    const DFS = (str,length)=>{
       if(length > 5) return;
        set.add(str);
        for(let i=0; i<vowel.length; i++){
            DFS(str+vowel[i],length+1);
        }
    }
  
    DFS('',0)
    return [...set].indexOf(word)
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.ArrayList;
import java.util.List;

class Solution {

    String[] words = {"A", "E", "I", "O", "U"};
    List<String> index = new ArrayList<>();

    public int solution(String word) {
        dfs("");

        return index.indexOf(word);
    }

    private void dfs(String str) {
        if (str.length() > 5) {
            return;
        }
        
        index.add(str);
        
        for (int i = 0; i < words.length ; i++) {
            dfs(str+words[i]);
        }
    }
}